packages/flame/test/experimental/layout_component_test.dart

```diff
@@ -0,0 +1,287 @@
+      expect(firstText.absolutePosition.x, 200);
+      expect(firstText.position.x, -300);
+      expect(secondText.position.x, -250);
+      expect(thirdText.position.x, -150);
+    });
+
+    testWithFlameGame('mainAxisAlignment.end with position, size and gap',
+        (game) async {
+      await _initScene(game, MainAxisAlignment.end, Vector2(500, 50), 20);
+      expect(firstText.absolutePosition.x, 160);
+      expect(firstText.position.x, -340);
+      expect(secondText.position.x, -270);
+      expect(thirdText.position.x, -150);
+    });
+
+    testWithFlameGame('mainAxisAlignment.spaceBetween with position and size',
+        (game) async {
+      await _initScene(game, MainAxisAlignment.spaceBetween, Vector2(50, 50));
+      final gap = (rowComponent.size.x - componentsWidth) / 2;
+      expect(secondText.position.x, gap + 50);
+      expect(thirdText.position.x, 2 * gap + 150);
+    });
+
+    testWithFlameGame('mainAxisAlignment.spaceEvenly with position and size',
+        (game) async {
+      await _initScene(game, MainAxisAlignment.spaceEvenly, Vector2(50, 50));
+      final gap = (rowComponent.size.x - componentsWidth) / 4;
+      expect(firstText.position.x, gap);
+      expect(secondText.position.x, 2 * gap + 50);
+      expect(thirdText.position.x, 3 * gap + 150);
+    });
+
+    testWithFlameGame('mainAxisAlignment.spaceAround with position and size',
+        (game) async {
+      await _initScene(game, MainAxisAlignment.spaceAround, Vector2(50, 50));
+      final gap = (rowComponent.size.x - componentsWidth) / 3;
+      expect(
+        double.parse((firstText.position.x).toStringAsFixed(1)),
+        double.parse((gap / 2).toStringAsFixed(1)),
+      );
+      expect(
+        double.parse((secondText.position.x).toStringAsFixed(1)),
+        double.parse((1.5 * gap + 50).toStringAsFixed(1)),
+      );
+      expect(
+        double.parse((thirdText.position.x).toStringAsFixed(1)),
+        double.parse((2.5 * gap + 150).toStringAsFixed(1)),
+      );
+    });
+
+    testWithFlameGame('mainAxisAlignment.center with position and size',
+        (game) async {
+      await _initScene(game, MainAxisAlignment.center, Vector2(50, 50));
+      final startPosition = (rowComponent.size.x - componentsWidth) / 2;
+      expect(firstText.position.x, startPosition);
+      expect(secondText.position.x, startPosition + 50);
+      expect(thirdText.position.x, startPosition + 150);
+    });
+
+    testWithFlameGame('mainAxisAlignment.center with position, size and gap',
+        (game) async {
+      await _initScene(game, MainAxisAlignment.center, Vector2(50, 50), 10);
+      final startPosition = (rowComponent.size.x - componentsWidth - 20) / 2;
+      expect(firstText.absolutePosition.x, startPosition + 50);
+      expect(firstText.position.x, startPosition);
+      expect(secondText.position.x, startPosition + 60);
+      expect(thirdText.position.x, startPosition + 170);
+    });
+  });
+
+  group('ColumnComponent', () {
```

Comment:
Seems like there is a lot of repetition here with the `RowComponent` - how about we merge them? Something like
```dart
for (var k = 0; k <= 1; k++) {
  group(k == 0 ? 'RowComponent' : 'ColumnComponent', () {
    Future<void> initScene(...) async {
      game.add(k == 0? RowComponent(...) : ColumnComponent(...));
    }

    testWithFlameGame('...', (game) async {
      ...
      expect(firstComponent.position[k], 0);
      expect(secondComponent.position[k], 60);
      expect(thirdComponent.position[k], 170);
    });
  });
}
```